src/labeler.ts

```diff
@@ -141,23 +146,65 @@ function getLabelGlobMapFromObject(
   return labelGlobs;
 }
 
-function toMatchConfig(config: StringOrMatchConfig): MatchConfig {
-  if (typeof config === 'string') {
+function toChangedFilesMatchConfig(config: any): ChangedFilesMatchConfig {
+  if (!config['changed-files']) {
+    return {};
+  }
+  const changedFilesConfig = config['changed-files'];
+
+  // If the value provided is a string or an array of strings then default to `any` matching
+  if (typeof changedFilesConfig === 'string') {
     return {
-      any: [config]
+      changedFiles: {
+        any: [changedFilesConfig]
+      }
     };
   }
 
-  return config;
+  const changedFilesMatchConfig = {
+    changedFiles: {}
+  };
+
+  if (Array.isArray(changedFilesConfig)) {
+    if (changedFilesConfig.every(entry => typeof entry === 'string')) {
+      changedFilesMatchConfig.changedFiles = {
+        any: changedFilesConfig
+      };
+    } else {
+      // If it is not an array of strings then it should be array of further config options
+      // so assign them to our `changedFilesMatchConfig`
+      changedFilesConfig.forEach(config => {
+        // Make sure that the values that we assign to our match config are an array
+        Object.entries(config).forEach(([key, value]) => {
+          const element = {
+            [key]: Array.isArray(value) ? value : [value]
+          };
+          Object.assign(changedFilesMatchConfig.changedFiles, element);
+        });
+      });
+    }
+  }
+
+  return changedFilesMatchConfig;
 }
 
-function printPattern(matcher: IMinimatch): string {
+function toMatchConfig(config: MatchConfig): MatchConfig {
+  const changedFilesConfig = toChangedFilesMatchConfig(config);
+  const branchConfig = toBranchMatchConfig(config);
+
+  return {
+    ...changedFilesConfig,
+    ...branchConfig
+  };
+}
+
+function printPattern(matcher: Minimatch): string {
   return (matcher.negate ? '!' : '') + matcher.pattern;
 }
 
 export function checkGlobs(
   changedFiles: string[],
-  globs: StringOrMatchConfig[]
+  globs: MatchConfig[]
```

Comment:
Currently, we can only specify one config for a label (`changed-files` or `base-branch` or `head-branch`). It would be great if we could provide more flexible behavior so that users can combine configs. The following update should solve it:
```ts
export function checkGlobs(
  changedFiles: string[],
  globs: MatchConfig[]
): boolean {
  for (const glob of globs) {
    core.debug(` checking pattern ${JSON.stringify(glob)}`);
    const matchConfig = toMatchConfig(glob);
    if (!checkMatch(changedFiles, matchConfig)) {
      return false;
    }
  }
  return true;
}
```